packages/flutter/test/widgets/multi_view_testing.dart

```diff
@@ -31,3 +31,48 @@ class FakeView extends TestFlutterView {
     // not produce consistent semantics trees.
   }
 }
+
+/// A test platform dispatcher that can show/hide its underlying `implicitView`,
+/// depending on the value of the [implicitViewHidden] flag.
+class NoImplicitViewPlatformDispatcher extends TestPlatformDispatcher {
+  NoImplicitViewPlatformDispatcher({ required super.platformDispatcher }) : superPlatformDispatcher = platformDispatcher;
+
+  final PlatformDispatcher superPlatformDispatcher;
+
+  bool implicitViewHidden = false;
+
+  @override
+  TestFlutterView? get implicitView {
+    return implicitViewHidden
+        ? null
+        : superPlatformDispatcher.implicitView as TestFlutterView?;
+  }
+}
+
+/// Test Flutter Bindings that allow tests to hide/show the `implicitView`
+/// of their [NoImplicitViewPlatformDispatcher] `platformDispatcher`.
+///
+/// This is used to test that [runApp] throws an assertion error with an
+/// explanation when used when the `implicitView` is disabled (like in Flutter
+/// web when multi-view is enabled).
+///
+/// Because of how [testWidgets] uses `runApp` internally to manage the lifecycle
+/// of a test, the implicitView must be disabled/reenabled inside of the body of
+/// the [WidgetTesterCallback] under test. In practice: the implicitView is disabled
+/// in the first line of the test, and reenabled in the last.
+///
+/// See: multi_view_no_implicitView_binding_test.dart
+class NoImplicitViewWidgetsBinding extends AutomatedTestWidgetsFlutterBinding {
+  late final NoImplicitViewPlatformDispatcher _platformDispatcher = NoImplicitViewPlatformDispatcher(platformDispatcher: super.platformDispatcher);
```

Comment:
Why `late`?

Reply:
Good question!

It seems that `late` allows me to use `super` when initializing the value!

(I copied this pattern from elsewhere in tests, but it's a smart interaction between `late` and `super`, wow!)

Reply:
> I guess the question is whether you want to terminate the app here in production or whether you want to keep running?

Yeah, an assertion here doesn't make much sense, since this is an un-recoverable error!